src/nvim/ops.c

```diff
@@ -2524,6 +2528,46 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+{
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST)) {
+    // No autocommand was defined
+    return;
+  }
+
+  // set v:event to a dictionary with information about the yank
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+
+  // the yanked text
+  list_T *list = list_alloc();
+  for (linenr_T i = 0; i < reg->y_size; i++) {
+    list_append_string(list, reg->y_array[i], -1);
+  }
+  list->lv_lock = VAR_FIXED;
+  dict_add_list(dict, "regcontents", list);
```

Comment:
You can test `set_vim_var_dict()` effect by trying `let v:completed_item.word = ''`, which results in:

    E46: Cannot change read-only variable "v:completed_item.word = ''"

> you must not use …di_flags = … in this case because this clears out DI_FLAGS_ALLOC

@bfredl I assume @ZyX-I was saying here to use `|=` instead. `v:event` should be "deeply" readonly.